core/datasource/definition/definition.go

```diff
@@ -116,7 +125,7 @@ func (s *Definition) Content() interface{} {
 
 // FromProto tries to build the Definiition object
 // from the given proto object.
-func FromProto(protoConfig *vegapb.DataSourceDefinition) (common.DataSourceType, error) {
+func FromProto(protoConfig *vegapb.DataSourceDefinition, tm ...time.Time) (common.DataSourceType, error) {
```

Comment:
Why are we passing a variadic list of time.Time here?

Reply:
Because DataSourceDefinition is the top level data source object that contains all of the others, and here it needs also to handle the `timetrigger.SpecConfigurationFromProto(dtp, tm)` method which needs time provided in order to update the nextTrigger value

Reply:
yea but you need only one time isn't it? which is the current time.

Reply:
Yes it takes only the first position.
The idea was to have the option to send `nil`; when you reach the step of the timetrigger you have this
```
func SpecConfigurationFromProto(protoConfig *vegapb.DataSourceSpecConfigurationTimeTrigger, tm []time.Time) (SpecConfiguration, error) {
	if tm == nil || len(tm) < 1 {
		return SpecConfiguration{}, errors.ErrMissingTimeForSettingTriggerRepetition
	}
	if protoConfig == nil {
		return SpecConfiguration{}, nil
	}

	return SpecConfiguration{
		Triggers:   common.InternalTimeTriggersFromProto(protoConfig.Triggers, tm[0]), // Use one position for now
		Conditions: common.SpecConditionsFromProto(protoConfig.Conditions),
	}, nil
}
```